tests/test_ble_interface_core.py

```
@@ -4725,12 +4726,35 @@ class InvalidDiscoveryClient:
+    """DiscoveryManager should accept clients that expose only `_discover`."""
+    filtered_device = _create_ble_device("AA:BB:CC:DD:EE:FF", "Filtered")
+    discover_result = {
+        "filtered": (
+            filtered_device,
+            SimpleNamespace(service_uuids=[SERVICE_UUID]),
+        ),
+    }
+
+    class UnderscoreDiscoveryClient:
+        @staticmethod
+        def _discover(**_kwargs: Any) -> dict[str, Any]:
```

Ruff (0.15.5)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**_kwargs`

(ANN401)